File: IRemember/app/src/main/java/ru/goodibunakov/iremember/adapter/TabAdapter.java
package ru.goodibunakov.iremember.adapter;

import android.util.Log;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import org.jetbrains.annotations.NotNull;

import ru.goodibunakov.iremember.fragment.CurrentTaskFragment;
import ru.goodibunakov.iremember.fragment.DoneTaskFragment;

/**
 * Created by GooDi on 18.09.2017.
 */

public class TabAdapter extends FragmentStatePagerAdapter {

    private final int numberOfTabs;
    public final static int CURRENT_TASK_FRAGMENT_POSITION = 0;
    public final static int DONE_TASK_FRAGMENT_POSITION = 1;

    private final CurrentTaskFragment currentTaskFragment;
    private final DoneTaskFragment doneTaskFragment;

    public TabAdapter(FragmentManager fm, int numberOfTabs) {
        super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
        this.numberOfTabs = numberOfTabs;
        this.currentTaskFragment = new CurrentTaskFragment();
        this.doneTaskFragment = new DoneTaskFragment();
        Log.d("debug", "TabAdapter");
    }

    @NotNull
    @Override
    public Fragment getItem(int position) {
        if (position == 0) {
            return currentTaskFragment;
        }
        return doneTaskFragment;
    }

    @Override
    public int getCount() {
        return numberOfTabs;
    }
}

File: IRemember/app/src/main/java/ru/goodibunakov/iremember/presentation/view/adapter/TabAdapter.kt
package ru.goodibunakov.iremember.presentation.view.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import ru.goodibunakov.iremember.presentation.view.fragment.CurrentTaskFragment
import ru.goodibunakov.iremember.presentation.view.fragment.DoneTaskFragment

class TabAdapter : FragmentStatePagerAdapter {

    companion object {
        const val CURRENT_TASK_FRAGMENT_POSITION = 0
        const val DONE_TASK_FRAGMENT_POSITION = 1
    }

    private val numberOfTabs: Int
    private val currentTaskFragment: CurrentTaskFragment
    private val doneTaskFragment: DoneTaskFragment

    constructor(fm: FragmentManager, numberOfTabs: Int): super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
        this.numberOfTabs = numberOfTabs
        this.currentTaskFragment = CurrentTaskFragment()
        this.doneTaskFragment = DoneTaskFragment()
    }

    override fun getItem(position: Int): Fragment {
        return if (position == 0) {
            currentTaskFragment
        } else doneTaskFragment
    }

    override fun getCount(): Int {
        return numberOfTabs
    }
}